### PR TITLE
Add dedicated build for nflow-explorer-ng

### PR DESCRIPTION
**Pull request by jsyrjala**

### CODE REVIEW

File: .github/workflows/build-nflow-explorer-ng.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - '**/*.md'
+concurrency:
+  group: build-nflow-explorer-ng-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+permissions:
+  contents: read
+  checks: write
+  security-events: write
+jobs:
+  build-and-test-nflow-explorer-ng:
```
Comment (efonsell) [inline .github/workflows/build-nflow-explorer-ng.yaml:17]: I would put this job to the build.yaml with other jobs until duplicating the boilerplate around the job can be avoided.

File: .github/workflows/build-nflow-explorer.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Build and test nflow-explorer
```
Comment (efonsell) [inline .github/workflows/build-nflow-explorer.yaml:1]: I think we should drop nflow-explorer, so this would not be needed

**Review by efonsell — CHANGES_REQUESTED** [review]

File: .github/workflows/build-nflow-explorer-ng.yaml (hunk 1 above)
Comment (jsyrjala) [reply .github/workflows/build-nflow-explorer-ng.yaml:17]: Doing it in separate .yaml makes it possible to run it in parallel with main build, thus making build faster.

Comment (efonsell) [reply .github/workflows/build-nflow-explorer-ng.yaml:17]: Different jobs in same file are ran in parallel, so having it in a separate file does not make build any faster.

File: .github/workflows/build-nflow-explorer.yaml (hunk 2 above)
Comment (jsyrjala) [reply .github/workflows/build-nflow-explorer.yaml]: dropped

File: .github/workflows/build-nflow-explorer-ng.yaml (hunk 1 above)
Comment (jsyrjala) [reply .github/workflows/build-nflow-explorer-ng.yaml]: fixed

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -36,6 +36,25 @@ jobs:
...
+          java-version: 11
+          distribution: 'temurin'
+          cache: 'maven'
+      - name: Run build and unit tests
+        run: mvn -T1C --no-transfer-progress --color=always -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -am -pl nflow-explorer-ng test
+      - name: Publish JUnit test report
+        uses: scacap/action-surefire-report@v1
+        if: always()
+        with:
+          check_name: 'Unit Test Report for nflow-explorer-ng'
```
Comment (efonsell) [inline .github/workflows/build.yaml:57]: This is failing as there are no results found, should this be removed?

Comment (jsyrjala) [reply .github/workflows/build.yaml]: Removed

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -31,11 +31,7 @@ jobs:
           cache: 'maven'
       - name: Run build and unit tests
         run: mvn -T1C --no-transfer-progress --color=always -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -am -pl nflow-jetty,nflow-netty,nflow-metrics test
-      - name: Publish JUnit test report
-        uses: scacap/action-surefire-report@v1
-        if: always()
-        with:
-          check_name: 'Unit Test Report for Java ${{ matrix.java }}'
```
Comment (efonsell) [inline .github/workflows/build.yaml]: Seems to be removed from java build, should be removed from explorer build?

**Review by efonsell — APPROVED** [review]

**Pull request merged**